common/containers/Tabs/Swap/components/wantToSwapMy.js

```diff
@@ -0,0 +1,167 @@
+    onChange: PropTypes.any,
+    kindOptions: PropTypes.any
+  };
+
+  render() {
+    return (
+      <span className="dropdown">
+        <select
+          value={this.props.kind}
+          className="btn btn-default"
+          onChange={this.props.onChange.bind(this)}
+        >
+          {this.props.kindOptions.map((obj, i) => {
+            return <option value={obj} key={i}>{obj}</option>;
+          })}
+        </select>
+      </span>
+    );
+  }
+}
+
+export default class WantToSwapMy extends Component {
+  constructor(props, context) {
+    super(props, context);
+  }
+
+  static propTypes = {
+    bityRates: PropTypes.any,
+    originAmount: PropTypes.any,
+    destinationAmount: PropTypes.any,
+    originKind: PropTypes.string,
+    destinationKind: PropTypes.string,
+    destinationKindOptions: PropTypes.array,
+    originKindOptions: PropTypes.array,
+    swapOriginKindTo: PropTypes.func,
+    swapDestinationKindTo: PropTypes.func,
+    swapOriginAmountTo: PropTypes.func,
+    swapDestinationAmountTo: PropTypes.func,
+    swapOriginKindAndDestinationKindAndDestinationOptionsTo: PropTypes.func
+  };
+
+  onClickStartSwap() {}
+
+  onChangeOriginAmount = amount => {
+    let originAmountAsNumber = parseFloat(amount);
+    if (originAmountAsNumber) {
+      let pairName = combineAndUpper(
+        this.props.originKind,
+        this.props.destinationKind
+      );
+      let bityRate = this.props.bityRates[pairName];
+      this.props.swapOriginAmountTo(originAmountAsNumber);
+      this.props.swapDestinationAmountTo(originAmountAsNumber * bityRate);
+    } else {
+      this.props.swapOriginAmountTo('');
+      this.props.swapDestinationAmountTo('');
+    }
+  };
+
+  onChangeDestinationAmount(amount) {
+    let destinationAmountAsNumber = parseFloat(amount);
+    if (destinationAmountAsNumber) {
+      this.props.swapDestinationAmountTo(destinationAmountAsNumber);
+      let pairName = combineAndUpper(
+        this.props.destinationKind,
+        this.props.originKind
+      );
+      let bityRate = this.props.bityRates[pairName];
+      this.props.swapOriginAmountTo(destinationAmountAsNumber * bityRate);
+    } else {
+      this.props.swapOriginAmountTo('');
+      this.props.swapDestinationAmountTo('');
+    }
+  }
+
+  async onChangeDestinationKind(event) {
+    let newDestinationKind = event.target.value;
+    this.props.swapDestinationKindTo(newDestinationKind);
+    let pairName = combineAndUpper(this.props.originKind, newDestinationKind);
+    let bityRate = this.props.bityRates[pairName];
+    this.props.swapDestinationAmountTo(
+      parseFloat(this.props.originAmount) * bityRate
+    );
+  }
+
+  async onChangeOriginKind(event) {
+    let newOriginKind = event.target.value;
+    this.props.swapOriginKindTo(newOriginKind);
+    // https://github.com/reactjs/redux/issues/1543#issuecomment-201399259
+    let store = window.store;
+    let destinationKind = store.getState().swap.destinationKind;
```

Comment:
Correct way to fix this is to move all depended state form this component to redux, or to look for props changes in componentwillupdate. Right now you can get inconsistent state if someone dispatches that action from somewhere else.